bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java

```diff
@@ -212,6 +214,44 @@ public void testJITInjection()
         assertEquals(stateLog, ImmutableList.of("postDependentInstance", "preDependentInstance"));
     }
 
+    @Test
+    public void testPreDestroyExceptionHandled()
+            throws Exception
+    {
+        Injector injector = Guice.createInjector(
+                Stage.PRODUCTION,
+                new LifeCycleModule(),
+                binder -> {
+                    binder.bind(DependentInstance.class).in(Scopes.SINGLETON);
+                    binder.bind(DestroyExceptionInstance.class).in(Scopes.SINGLETON);
+                });
+
+        LifeCycleManager lifeCycleManager = injector.getInstance(LifeCycleManager.class);
+        lifeCycleManager.start();
+        assertEquals(stateLog, ImmutableList.of("postDependentInstance", "postDestroyException"));
+        try {
+            lifeCycleManager.stop();
+            fail("Expected exception to be thrown");
+        }
+        catch (LifeCycleStopException e) {
+            assertEquals(e.getSuppressed().length, 2, "Expected two suppressed exceptions");
+            Set<String> suppressedMessages = Arrays.stream(e.getSuppressed())
+                    .map(Throwable::getMessage)
+                    .collect(Collectors.toSet());
+            assertEquals(suppressedMessages, Sets.newHashSet("preDestroyExceptionOne", "preDestroyExceptionTwo"));
+        }
+        catch (Exception e) {
+            fail("Expected LifeCycleStopException to be throw, found: " + e.getClass());
+        }
+
+        assertEquals(new HashSet<>(stateLog), Sets.newHashSet(
```

Comment:
`ImmutableSet.copyOf` and `ImmutableSet.of`

Reply:
Will change.